File: src/main/java/shootemup/geometry/Range.java
package shootemup.geometry;


public class Range implements IRange {

    public Range(int lower, int upper) {
	this._lower = lower;
	this._upper = upper;
    }

    @Override
    public boolean contains(int value) {
	return value >= _lower && value < _upper;
    }

    // These are only public to ease testing.
    // Do not access these; they are implementation details.
    public final int _lower;
    public final int _upper;
}

File: src/main/kotlin/shootemup/geometry/Range.kt
package shootemup.geometry

// Member variables are only public to ease testing.
// Do not access these; they are implementation details.
class Range(val _lower: Int, val _upper: Int) : IRange {

    override fun contains(value: Int): Boolean {
        return value >= _lower && value < _upper
    }
}
